conans/client/graph/graph_builder.py

```diff
@@ -246,13 +245,12 @@ def _create_new_node(self, node, require, graph, profile_host, profile_build, gr
                                      "host dependency")
             require.ref.version = transitive.require.ref.version
 
+        resolved = self._resolved_system(node, require, profile_build, profile_host,
```

Comment:
I like the change for `platform_requires` to now be lockable (if Im reading the change right!), maybe we should add a changelog entry for that too as a bugfix?

At the very least, I think that at leats one test should now check that the lock properly works, just in case :)

Reply:
I think the behavior before was buggy, the relevant change was in ``def test_system_tool_require_range(self):`` (this is the reason I didn't refactor the names in this test @franramirez688, to not hide this change in the diff)